inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/InlongClusterServiceImpl.java

```diff
@@ -169,42 +169,41 @@ public Boolean updateTag(ClusterTagRequest request, String operator) {
             LOGGER.warn("inlong cluster tag was not exist for id={}", id);
             return true;
         }
-        InlongClusterTagEntity tagExist = clusterTagMapper.selectByTag(newClusterTag);
-        if (tagExist != null) {
-            String errMsg = String.format("inlong cluster tag [%s] already exist", newClusterTag);
-            LOGGER.error(errMsg);
-            throw new BusinessException(errMsg);
-        }
 
-        // check if there are some InlongGroups that uses this tag
         String oldClusterTag = exist.getClusterTag();
-        this.assertNoInlongGroupExists(oldClusterTag);
+        if (!newClusterTag.equals(oldClusterTag)) {
+            InlongClusterTagEntity tagConflict = clusterTagMapper.selectByTag(newClusterTag);
+            if (tagConflict != null) {
+                String errMsg = String.format("inlong cluster tag [%s] already exist by id [%d]", newClusterTag,
```

Comment:
The cluster tag is enough, no need to record the id because the id was passed from outsides.

Reply:
The id is the other name-conflicting tag entity id, not the one passed from client.